File: scripts/import-medicines.py
import csv
import json
import sys

def convert_csv_to_json(csv_path, output_path):
    medicines = []
    
    with open(csv_path, 'r', encoding='utf-8') as f:
        reader = csv.DictReader(f)
        for row in reader:
            medicine = {
                "name": row['药品名称'].strip(),
                "code": row['简码'].strip().upper(),
                "price": float(row['单价']) if row['单价'] else 0,
                "unit": row['单位'].strip() if row['单位'] else 'g',
                "dosage": float(row['常用用量']) if row['常用用量'] and row['常用用量'].replace('.', '').isdigit() else 10,
                "category": "",
                "costPrice": float(row['进价']) if row['进价'] else 0,
                "stock": int(row['库存']) if row['库存'] else 0
            }
            medicines.append(medicine)
    
    output_data = {
        "medicines": medicines,
        "exportDate": "2026-05-08",
        "description": "600味中药库数据"
    }
    
    with open(output_path, 'w', encoding='utf-8') as f:
        json.dump(output_data, f, ensure_ascii=False, indent=2)
    
    print(f"成功转换 {len(medicines)} 种药品")
    print(f"输出文件: {output_path}")

if __name__ == "__main__":
    csv_path = "../中药库_2026-5-6.csv"
    output_path = "../中药库数据.json"
    convert_csv_to_json(csv_path, output_path)

File: scripts/import-medicines.js
const fs = require('fs');
const path = require('path');

const csvPath = path.join(__dirname, '../../中药库_2026-5-6.csv');
const outputPath = path.join(__dirname, '../中药库数据.json');

const content = fs.readFileSync(csvPath, 'utf8');
const lines = content.split('\n').filter(line => line.trim());

const headers = lines[0].split(',');
const medicines = [];

for (let i = 1; i < lines.length; i++) {
    const values = lines[i].split(',');
    const medicine = {};
    
    headers.forEach((header, index) => {
        medicine[header.trim()] = values[index] ? values[index].trim() : '';
    });
    
    medicines.push({
        name: medicine['药品名称'] || '',
        code: (medicine['简码'] || '').toUpperCase(),
        price: parseFloat(medicine['单价']) || 0,
        unit: medicine['单位'] || 'g',
        dosage: !isNaN(parseFloat(medicine['常用用量'])) ? parseFloat(medicine['常用用量']) : 10,
        category: '',
        costPrice: parseFloat(medicine['进价']) || 0,
        stock: parseInt(medicine['库存']) || 0
    });
}

const outputData = {
    medicines: medicines,
    exportDate: new Date().toISOString().split('T')[0],
    description: `${medicines.length}味中药库数据`
};

fs.writeFileSync(outputPath, JSON.stringify(outputData, null, 2), 'utf8');

console.log(`成功转换 ${medicines.length} 种药品`);
console.log(`输出文件: ${outputPath}`);